package.json

```diff
@@ -3,16 +3,21 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@reduxjs/toolkit": "^1.9.7",
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^13.0.0",
     "@testing-library/user-event": "^13.2.1",
+    "axios": "^1.6.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-redux": "^8.1.3",
+    "react-router-dom": "^6.20.0",
     "react-scripts": "5.0.1",
+    "redux": "^4.2.1",
     "web-vitals": "^2.1.0"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "PORT=3001 react-scripts start",
     "build": "react-scripts build",
```

Comment:
- [ ] While you have done a good job for both the Front and backend, the frontend could not start because of this that you added. I understand that running two apps would make the api run first and from the localhost 3000 which means you think there is not localhost 3000 for your react to run, kindly note that once the rails starts running on localhost 3000, when you run for the react, it would know that another app is running on host 3000 and it will ask you if it should go on to 3001, so you will type `y` which means yes. Do you understand, please?